user.py

```diff
@@ -100,7 +100,7 @@ def is_allowed_user(user_id):
 
 @user_bp.route('/profile/<int:user_id>/<int:current_year>')
 @login_required
-@check_access_level(Role.ADMIN, Role.RESEARCHER, Role.TEACHER)
+@check_access_level(Role.USER)
```

Comment:
There is no need to define such a role. You can simply allow any logged user and remove this decorator.